src/util/StringUtils.h

```diff
@@ -291,11 +318,26 @@ static std::string listToString(const Range& r,
   // Add all the string representations to the stream with seperator betwenn
   // them.
   std::ranges::for_each_n(rAsString.begin(), rAsString.size() - 1,
-                          [&stream, &listItemSeparator](const auto& listItem) {
-                            stream << listItem << listItemSeparator;
+                          [&stream, &separator](const auto& listItem) {
+                            (*stream) << listItem << separator;
```

Comment:
This function currently assumest that the size of the range is known in advance. 
This limits its functionality (The range cannot be lazily created). Change this, s.t. you can also pass in a lazy range.
("Add the separator before every element that is not the first" is the pattern to go here).
Then also add a unit test for such a lazy range (e.g. something from a Generator etc.)

Reply:
Done as of [714c677](https://github.com/ad-freiburg/qlever/pull/1010/commits/714c6772fabc9fc1332aa20302eb3c94af419fa9), but I'm still hunting bugs.

Reply:
Fixed the last (known) bug as of [0baaf4d](https://github.com/ad-freiburg/qlever/pull/1010/commits/0baaf4d9d3ab31960f1102c831cd48e300dc16a5).